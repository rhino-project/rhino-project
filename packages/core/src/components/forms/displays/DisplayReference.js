import { useCallback } from 'react';
import { useGlobalComponent } from '../../../hooks/overrides';
import { DisplayString } from './DisplayString';

export const DisplayReferenceBase = (props) => {
  const accessor = useCallback((value) => value?.display_name, []);

  return <DisplayString {...props} accessor={accessor} />;
};

export const DisplayReference = (props) =>
  useGlobalComponent('DisplayReference', DisplayReferenceBase, props);
