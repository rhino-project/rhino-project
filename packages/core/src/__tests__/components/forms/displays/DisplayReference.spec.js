import { sharedDisplayTests } from './sharedDisplayTests';
import DisplayReference from 'rhino/components/forms/displays/DisplayReference';

describe('DisplayReference', () => {
  sharedDisplayTests(DisplayReference);
});
