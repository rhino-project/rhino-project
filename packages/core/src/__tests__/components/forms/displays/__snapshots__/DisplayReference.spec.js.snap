// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayReference > renders inside of model context 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="-"
  />
</DocumentFragment>
`;

exports[`DisplayReference > should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="-"
  />
</DocumentFragment>
`;

exports[`DisplayReference > should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="baz"
  />
</DocumentFragment>
`;

exports[`DisplayReference > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
